src/Sets/Ball2.jl

```diff
@@ -259,12 +262,12 @@ Random sampling with uniform distribution in `B` is computed using Muller's meth
 of normalized Gaussians. This function requires the package `Distributions`.
 See `_sample_unit_nball_muller!` for implementation details.
 """
-function sample(B::Ball2{N}, nsamples::Int=1;
+function sample(B::Ball2{N, VN}, nsamples::Int=1;
                 rng::AbstractRNG=GLOBAL_RNG,
-                seed::Union{Int, Nothing}=nothing) where {N<:AbstractFloat}
+                seed::Union{Int, Nothing}=nothing) where {N<:AbstractFloat, VN<:AbstractVector{N}}
     require(:Distributions; fun_name="sample")
     n = dim(B)
-    D = Vector{Vector{N}}(undef, nsamples) # preallocate output
+    D = Vector{VN}(undef, nsamples) # preallocate output
```

Comment:
You cannot know that the samples will have the same vector type as the set.

Reply:
That's weird.. but possible. So you suggest to hardcode to vector?

Reply:
i checked and https://github.com/JuliaReach/LazySets.jl/blob/master/src/Utils/samples.jl#L187 is only defined for cache D with Vector. this means that `D = Vector{VN}(undef, nsamples)` will not work if `VN` is different from vector. ill open a separate issue, and revert my change here.